packages/opentelemetry-exporter-collector/src/types.ts

```diff
@@ -0,0 +1,482 @@
+   * crosses a process boundary. True when the parent_span belongs to the same
+   * process as the current span.
+   */
+  sameProcessAsParentSpan?: boolean;
+
+  //@TODO - do we use it in opentelemetry or it is not needed?
+  // /**
+  //  * An optional number of child spans that were generated while this span was
+  //  * active. If set, allows an implementation to detect missing child spans.
+  //  */
+  // childSpanCount?: number;
+  /**
+   * The included links.
+   */
+  links?: Links;
+}
+
+/**
+ * A single stack frame in a stack trace.
+ */
+export interface StackFrame {
+  /**
+   * The fully-qualified name that uniquely identifies the function or method
+   * that is active in this frame.
+   */
+  functionName?: TruncatableString;
+  /**
+   * An un-mangled function name, if `function_name` is
+   * [mangled](http://www.avabodh.com/cxxin/namemangling.html). The name can be
+   * fully qualified.
+   */
+  originalFunctionName?: TruncatableString;
+  /**
+   * The name of the source file where the function call appears.
+   */
+  fileName?: TruncatableString;
+  /**
+   * The line number in `file_name` where the function call appears.
+   */
+  lineNumber?: string;
+  /**
+   * The column number where the function call appears, if available. This is
+   * important in JavaScript because of its anonymous functions.
+   */
+  columnNumber?: string;
+  /**
+   * The binary module from where the code was loaded.
+   */
+  loadModule?: Module;
+  /**
+   * The version of the deployed source code.
+   */
+  sourceVersion?: TruncatableString;
+}
+
+/**
+ * A collection of stack frames, which can be truncated.
+ */
+export interface StackFrames {
+  /**
+   * Stack frames in this call stack.
+   */
+  frame?: StackFrame[];
+  /**
+   * The number of stack frames that were dropped because there were too many
+   * stack frames. If this value is 0, then no stack frames were dropped.
+   */
+  droppedFramesCount?: number;
+}
+
+/**
+ * The call stack which originated this span.
+ */
+export interface StackTrace {
+  /**
+   * Stack frames in this stack trace.
+   */
+  stackFrames?: StackFrames;
+  /**
+   * The hash ID is used to conserve network bandwidth for duplicate stack
+   * traces within a single trace.  Often multiple spans will have identical
+   * stack traces. The first occurrence of a stack trace should contain both
+   * `stack_frames` and a value in `stack_trace_hash_id`.  Subsequent spans
+   * within the same request can refer to that stack trace by setting only
+   * `stack_trace_hash_id`.
+   */
+  stackTraceHashId?: string;
+}
+
+/**
+ * A time-stamped annotation or message event in the Span.
+ */
+export interface TimeEvent {
+  /**
+   * The time the event occurred.
+   */
+  time?: string;
+  /**
+   * A text annotation with a set of attributes.
+   */
+  annotation?: Annotation;
+  /**
+   * An event describing a message sent/received between Spans.
+   */
+  messageEvent?: MessageEvent;
```

Comment:
Looks like `MessageEvent` interface is not defined anywhere, could you please help me to point out the same?

Reply:
looks like I have missed this interface

Reply:
@mayurkale22 ` ... open an issue to define the semantic conversion ...` are you talking about raising issue in `opentelemetry-js` or in `opentelemetry-specification`? Is there already a definition of attributes which I can check against ?

Reply:
> looks like I have missed this interface

Looks like currently this refers to Web APIs Message Event: https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent

Reply:
I am ok with https://github.com/open-telemetry/opentelemetry-js/commit/2b67bed3075b81ae9dc271a1901b901dfac40787, thanks for addressing it.